utils/kubernetes/kubeutil.py

```diff
@@ -552,7 +552,24 @@ def _fetch_host_data(self):
                 # if not found, use an empty string - we use None as "not initialized"
                 self._node_ip = status.get('hostIP', '')
                 self._node_name = spec.get('nodeName', '')
-                break
+                return
+
+        # At this point the agent isn't inside the podList.
+        # Take the first Pod with a status:
+        # all running pods have the adapted '.spec.nodeName'
+        # static pods doesn't have the '.status.hostIP'
```

Comment:
Do we need to keep the previous logic then? It looks like your block could be a replacement instead of an addition.

Reply:
@hkaj I removed the old block and keep the associated test to confirm this change.